include/nil/crypto3/zk/snark/systems/plonk/placeholder/permutation_argument.hpp

```diff
@@ -113,15 +110,22 @@ namespace nil {
                         math::polynomial_dfs<typename FieldType::value_type> V_P(basic_domain->size() - 1,
                                                                                  basic_domain->size());
 
-                        std::vector<math::polynomial_dfs<typename FieldType::value_type>> g_v;
-                        std::vector<math::polynomial_dfs<typename FieldType::value_type>> h_v;
+                        std::vector<math::polynomial_dfs<typename FieldType::value_type>> g_v(S_id.size());
+                        std::vector<math::polynomial_dfs<typename FieldType::value_type>> h_v(S_id.size());
                         for (std::size_t i = 0; i < S_id.size(); i++) {
                             BOOST_ASSERT(column_polynomials[i].size() == basic_domain->size());
                             BOOST_ASSERT(S_id[i].size() == basic_domain->size());
                             BOOST_ASSERT(S_sigma[i].size() == basic_domain->size());
 
-                            g_v.push_back(column_polynomials[i] + beta * S_id[i] + gamma);
-                            h_v.push_back(column_polynomials[i] + beta * S_sigma[i] + gamma);
+                            g_v[i] = S_id[i];

```

Comment:
Please add the old code as a comment, so we can read it easier, like
// g_v.push_back(column_polynomials[i] + beta * S_id[i] + gamma);

Reply:
updated in 3a158132